tests/js/server-tests.js

```diff
@@ -0,0 +1,169 @@
+
+    jqUnit.assertNotNull("There should be a body.", body);
+};
+
+testServer.runTests = function () {
+    jqUnit.module("Tests for inlining of templates...");
+
+    jqUnit.asyncTest("Confirm that template content is inlined...", function () {
+        request.get(testServer.options.config.express.baseUrl + "inline", function (error, response, body) {
+            jqUnit.start();
+
+            testServer.isSaneResponse(jqUnit, error, response, body);
+
+            if (body) {
+                var templateRegexp = /type="text\/x-handlebars-template">/gi;
+                var matches = body.match(templateRegexp);
+
+                jqUnit.assertNotNull("There should be templates found in the body.", matches);
+                if (matches) {
+                    // TODO:  This will need to be updated as we add templates.  If it comes up often, refactor.
+                    jqUnit.assertTrue("There should be at least three templates in the returned source.", matches.length > 3);
+                }
+            }
+        });
+    });
+
+    jqUnit.module("Tests for dispatcher...");
+
+    // Test the following variations:
+    //
+    //   1. The default home page with the default layout
+    //   2. A custom page with matching layout
+    //   3. A custom page with no matching layout
+    //
+    //  All variations should test partials and variables
+    var pages = ["custom", "custom-no-matching-layout"];
+
+    pages.forEach(function (page) {
+        jqUnit.asyncTest("Test template handling dispatcher for page '" + page + "' ...", function () {
+            request.get(testServer.options.config.express.baseUrl + "dispatcher/" + page + "?myvar=queryvariable", function (error, response, body) {
+                jqUnit.start();
+
+                testServer.isSaneResponse(jqUnit, error, response, body);
+
+                jqUnit.assertNotNull("There should be layout content in the body...",         body.match(/from the layout/));
+                jqUnit.assertNotNull("There should be page content in the body...",           body.match(/from the page/));
+                jqUnit.assertNotNull("There should be partial content in the body...",        body.match(/from the partial/));
+
+                var mdRegexp = /<p><em>this works<\/em><\/p>/i;
+                jqUnit.assertNotNull("The results should contain transformed markdown.", body.match(mdRegexp));
+
+                jqUnit.assertNotNull("There should be model variable content in the body...", body.match(/modelvariable/));
+                jqUnit.assertNotNull("There should be query variable content in the body...", body.match(/queryvariable/));
+
+                // Tests for the "equals" helper
+
+                // TODO:  If I end up using this pattern any more often, make it into a function
+                // TODO:  Also, check with Antranig about jQuery like find, etc. functionality in fluid itself.
+                var equalsElementRegexp = /<td class="equal">([^<]+)<\/td>/;
```

Comment:
Yes, it did feel dirty to do it that way.  I may look at what Zombie uses for DOM parsing and convert for that for future non-browser "did I receive markup with one of these elements" kind of tests.